IPlug/CLAP/IPlugCLAP.h

```diff
@@ -0,0 +1,250 @@
+/*
+ ==============================================================================
+ 
+ This file is part of the iPlug 2 library. Copyright (C) the iPlug 2 developers. 
+ 
+ See LICENSE.txt for  more info.
+ 
+ ==============================================================================
+*/
+
+#ifndef _IPLUGAPI_
+#define _IPLUGAPI_
+// Only load one API class!
+
+/**
+ * @file
+ * @copydoc IPlugCLAP
+ */
+
+#include "IPlugAPIBase.h"
+#include "IPlugProcessor.h"
+#include "plugin.hh"
+
+#include "config.h"   // This is your plugin's config.h.
+
+BEGIN_IPLUG_NAMESPACE
+
+/** Used to pass various instance info to the API class */
+struct InstanceInfo
+{
+  const clap_plugin_descriptor* mDesc;
+  const clap_host* mHost;
+};
+
+// Set the level of host checking based on if this is debug build
+
+#ifdef _DEBUG
+using ClapPluginHelper = clap::helpers::Plugin<clap::helpers::MisbehaviourHandler::Terminate, clap::helpers::CheckingLevel::Maximal>;
+using ClapHost = clap::helpers::HostProxy<clap::helpers::MisbehaviourHandler::Terminate, clap::helpers::CheckingLevel::Maximal>;
+#else
+using ClapPluginHelper = clap::helpers::Plugin<clap::helpers::MisbehaviourHandler::Ignore, clap::helpers::CheckingLevel::None>;
+using ClapHost = clap::helpers::HostProxy<clap::helpers::MisbehaviourHandler::Ignore, clap::helpers::CheckingLevel::None>;
+#endif
+
+/** CLAP API base class for an IPlug plug-in
+*   @ingroup APIClasses */
+class IPlugCLAP : public IPlugAPIBase
+                , public IPlugProcessor
+                , public ClapPluginHelper
+{
+  struct ParamToHost
+  {
+    enum class Type { Begin, Value, End };
+
+    ParamToHost()
+      : mType(Type::Value)
+      , mIdx(-1)
+      , mValue(0.0)
+    {}
+
+    ParamToHost(Type type, int idx, double value)
+      : mType(type)
+      , mIdx(idx)
+      , mValue(value)
+    {}
+
+    uint32_t idx() const { return static_cast<uint32_t>(mIdx); }
+    double value() const { return mValue; }
+    uint16_t type() const
+    {
+      switch (mType)
+      {
+        case Type::Begin:   return CLAP_EVENT_PARAM_GESTURE_BEGIN;
+        case Type::End:     return CLAP_EVENT_PARAM_GESTURE_END;
+        case Type::Value:   // fall through
+        default:            return CLAP_EVENT_PARAM_VALUE;
+      }
+    }
+    
+    Type mType;
+    int mIdx;
+    double mValue;
+  };
+  
+public:
+  IPlugCLAP(const InstanceInfo& info, const Config& config);
+
+  // IPlugAPIBase
+  void BeginInformHostOfParamChange(int idx) override;
+  void InformHostOfParamChange(int idx, double normalizedValue) override;
+  void EndInformHostOfParamChange(int idx) override;
+//  void InformHostOfPresetChange() override;
```

Comment:
TODO?

Reply:
Do you mean the InformHostOfPresetChange()?

Reply:
Yes, i removed it for now